### PR TITLE
Remove web_stories_is_poster attachment meta

### LINKED ISSUE #5638: Remove web_stories_is_poster  attachment meta
## Feature Description

Follow up from #5316 and #5635. 

Remove use of attachment meta web_stories_is_poster now that taxonomy term is used instead. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Remove saving to post meta, remove rows in database and clean out code. 

## Testing Instructions

---

Fixes #5638

### CODE REVIEW

File: includes/Database_Upgrader.php [hunk 1]
```diff
@@ -330,6 +333,31 @@ protected function yoast_reindex_stories() {
...
+				"SELECT post_id FROM $wpdb->postmeta WHERE meta_key = %s",
+				Media::POSTER_POST_META_KEY
+			)
+		);
+
+		if ( is_array( $post_ids ) && ! empty( $post_ids ) ) {
+			foreach ( $post_ids as $post_id ) {
+				delete_post_meta( (int) $post_id, Media::POSTER_POST_META_KEY );
+			}
+		}
```
Comment (swissspidy) [inline includes/Database_Upgrader.php]: Can just use `delete_post_meta_by_key( Media::POSTER_POST_META_KEY )`

Comment (spacedmonkey) [inline includes/Database_Upgrader.php]:
```suggestion
		delete_post_meta_by_key( Media::POSTER_POST_META_KEY );
```

**Comment by swissspidy** [comment]
#5635 and #5636 have been merged, so this needs a rebase

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
@spacedmonkey I think this is ready?

**Pull request merged**